packages/swapper/src/swappers/osmosis/utils/helpers.ts

```diff
@@ -0,0 +1,256 @@
+import { cosmos, osmosis } from '@shapeshiftoss/chain-adapters'
+import { bip32ToAddressNList, HDWallet } from '@shapeshiftoss/hdwallet-core'
+import axios from 'axios'
+import { find } from 'lodash'
+
+import { SwapError, TradeResult } from '../../../index'
+import { bn, bnOrZero } from '../../utils/bignumber'
+import { GAS, OSMOSIS_PRECISION } from './constants'
+import { IbcTransferInput, PoolInfo } from './types'
+
+export interface SymbolDenomMapping {
+  OSMO: string
+  ATOM: string
+  USDC: string
+}
+
+export const symbolDenomMapping = {
+  OSMO: 'uosmo',
+  ATOM: 'ibc/27394FB092D2ECCD56123C74F36E4C1F926001CEADA9CA97EA622B25F41E5EB2',
+  USDC: 'ibc/D189335C6E4A68B513C10AB227BF1C1D38C746766278BA3EEB4FB14124F1D858'
+}
+
+const txStatus = async (txid: string, baseUrl: string): Promise<string> => {
+  try {
+    const txResponse = await axios.get(`${baseUrl}/txs/${txid}`)
+    if (!txResponse?.data?.codespace && !!txResponse?.data?.gas_used) return 'success'
+    if (txResponse?.data?.codespace) return 'failed'
+  } catch (e) {
+    console.error('Failed to get status')
+  }
+  return 'not found'
+}
+
+export const pollForComplete = async (txid: string, baseUrl: string): Promise<string> => {
```

Comment:
maybe add a todo or create issue to see if we can leverage websockets provided by unchained for this